packages/cli/src/databases/repositories/user.repository.ts

```diff
@@ -125,15 +125,17 @@ export class UserRepository extends Repository<User> {
 		transactionManager?: EntityManager,
 	): Promise<{ user: User; project: Project }> {
 		const createInner = async (entityManager: EntityManager) => {
-			const newUser = entityManager.create(User, user);
-			const savedUser = await entityManager.save<User>(newUser);
-			const savedProject = await entityManager.save<Project>(
+			const newUser = user instanceof User ? user : entityManager.create(User, user);
```

Comment:
This checks if the user is already a `User` or just an object with the user properties.

Not doing this check would send TypeORM down an infinite loop because the `createUserFromSamlAttributes` creates a `User` and an `AuthIdentity` and assigns them to one another and vice versa:

https://github.com/n8n-io/n8n/blob/744e22042ac57ddb703dc9c1cb79c46f7f47cce8/packages/cli/src/sso/saml/samlHelpers.ts#L100-L101

https://github.com/n8n-io/n8n/blob/744e22042ac57ddb703dc9c1cb79c46f7f47cce8/packages/cli/src/sso/saml/samlHelpers.ts#L111-L111

https://github.com/n8n-io/n8n/blob/744e22042ac57ddb703dc9c1cb79c46f7f47cce8/packages/cli/src/sso/saml/samlHelpers.ts#L116-L116

Reply:
outdated: I removed the circular references